frontend/class-opengraph.php

```diff
@@ -649,4 +558,208 @@ public function publish_date() {
 		return true;
 	}
 
-} /* End of class */
\ No newline at end of file
+} /* End of class */
+
+class WPSEO_OpenGraph_Image {
+
+	private $options;
+
+	/**
+	 * @var array $images Holds the images that have been put out as OG image.
+	 */
+	private $images = array();
+
+	/**
+	 * Constructor
+	 *
+	 * @param array $options
+	 */
+	public function __construct( $options ) {
+		$this->options = $options;
+		$this->set_images();
+	}
+
+	/**
+	 * Return the images array
+	 *
+	 * @return array
+	 */
+	public function get_images() {
+		return $this->images;
+	}
+
+	/**
+	 * Check if page is front page or singular and call the corresponding functions. If not, call get_default_image.
+	 */
+	private function set_images() {
+		if ( is_front_page() ) {
+			$this->get_front_page_image();
+		}
+
+		if ( is_singular() ) {
+			$this->get_singular_image();
+		}
+
+		$this->get_default_image();
+	}
+
+	/**
+	 * If the frontpage image exists, call add_image
+	 */
+	private function get_front_page_image() {
+		if ( $this->options['og_frontpage_image'] !== '' ) {
+			$this->add_image( $this->options['og_frontpage_image'] );
+		}
+	}
+
+	/**
+	 * Get the images of the singular post.
+	 */
+	private function get_singular_image() {
+		global $post;
+
+		if ( $this->get_opengraph_image() ) {
+			return;
+		}
+
+		if ( $this->get_featured_image( $post->ID ) ) {
+			return;
+		}
+
+		$this->get_content_images( $post );
+	}
+
+	/**
+	 * Get default image and call add_image
+	 */
+	private function get_default_image() {
+		if ( count( $this->images ) == 0 && isset( $this->options['og_default_image'] ) && $this->options['og_default_image'] !== '' ) {
+			$this->add_image( $this->options['og_default_image'] );
+		}
+	}
+
+	/**
+	 * If opengraph-image is set, call add_image and return true
+	 *
+	 * @return bool
+	 */
+	private function get_opengraph_image() {
+		$ogimg = WPSEO_Meta::get_value( 'opengraph-image' );
+		if ( $ogimg !== '' ) {
+			$this->add_image( $ogimg );
+
+			return true;
+		}
+	}
+
+	/**
+	 * If there is a featured image, check image size. If image size is correct, call add_image and return true
+	 *
+	 * @param $post_id
+	 *
+	 * @return bool
+	 */
+	private function get_featured_image( $post_id ) {
+		if ( function_exists( 'has_post_thumbnail' ) && has_post_thumbnail( $post_id ) ) {
+			/**
+			 * Filter: 'wpseo_opengraph_image_size' - Allow changing the image size used for OpenGraph sharing
+			 *
+			 * @api string $unsigned Size string
+			 */
+			$thumb = wp_get_attachment_image_src( get_post_thumbnail_id( $post_id ), apply_filters( 'wpseo_opengraph_image_size', 'original' ) );
+
+			if ( $this->check_image_size( $thumb ) ) {
+				$this->add_image( $thumb[0] );
+
+				return true;
```

Comment:
But it's valid, we can return the value directly anyway. The return value was only ment to break out of the method. It could be changed.